keps/sig-autoscaling/20181106-in-place-update-of-pod-resources.md

```diff
@@ -0,0 +1,423 @@
+PodCondition's Status and Reason fields respectively.
+
+```text
+
+                                  +------------+
+                                  |4           |
+                       +----------+---------+  |
+                       |                    |  |
+            +----------+        false       <--+
+            |          | FailedNodeCapacity |
+            |          |                    |
+            |          +--------+---^-------+
+           1|                  5|   |
+ +----------v---------+         |   |
+ |                    |         |   |
+ |        true        |         |   |
+ | AwaitingPreemption |         |   |
+ |                    |         |   |
+ +-------^-----+------+         |   |
+        1|     |2               |   |4
+         |     |             +--v---+---+
+         |     |             |          |
+         |     +------------->  false   <--+
+         |                   | Accepted |  |
+         +-------------------+          |  |
+                             |          |  |
+                             +----+-----+  |
+                                  |3       |
+                                  +--------+
+
+```
+
+1. Scheduler, on starting pre-emption.
+1. Scheduler, after pre-emption, new resize request fits node.
+1. Scheduler, no pre-emption needed, new resize request fits node.
+1. Scheduler, node does not have capacity.
+1. Scheduler, no pre-emption needed, new resize request fits node.
+
+The following diagram shows possible transitions of PodResizeSuccess
+PodCondition's Status and Reason fields respectively.
+
+```text
+
+                                  +---------+
+                                  |         |
+            +---------------------+  true   <------------+
+            |                     | Success |            |
+            |                     |         |            |
+            |                     +----+----+            |
+           1|                         2|                 |
+ +----------v---------+      +---------v--------+        |
+ |                    |      |                  |        |
+ |        false       |      |       false      |        |
+ | FailedNodeCapacity |      | Rejected<Reason> |        |
+ |                    |      |                  |        |
+ +----------+---------+      +------------------+        |
+           3|                                            |
+            |                     +----------+           |
+            |                     |          |           |
+            +--------------------->  false   +-----------+
+                                  | Retrying |4
+                                  |          |
+                                  +----------+
+
+```
+
+1. Kubelet, if not enough space on Node.
+1. Any Controller, on permanent issue.
+1. Initiating actor (controller), on retry.
+1. On retry and successful resizing.
+
+#### Container resource limit update ordering
```

Comment:
below in possible extensions you mention allow limits to be updated, i am confused it his proposal is allowing both requests and limits to be updated or only one of them

Reply:
both